uvicorn/protocols/websockets/wsproto_impl.py

```diff
@@ -94,8 +131,9 @@ def resume_writing(self):
 
     def shutdown(self):
         self.queue.put_nowait({"type": "websocket.disconnect", "code": 1012})
-        self.conn.close(1012)
-        output = self.conn.bytes_to_send()
+        # self.conn.close(1012)
+        output = self.conn.send(wsproto.events.CloseConnection(code=code))
+        # output = self.conn.bytes_to_send()
```

Comment:
All these bits correspond to: "Switch from a bytes_to_send method to the send method returning the bytes to send directly. "